oauth2/oauth2-resource-server/src/main/java/org/springframework/security/oauth2/server/resource/authentication/JwtAuthenticationProvider.java

```diff
@@ -80,8 +82,10 @@ public Authentication authenticate(Authentication authentication) throws Authent
 		Jwt jwt;
 		try {
 			jwt = this.jwtDecoder.decode(bearer.getToken());
-		} catch (JwtException failed) {
+		} catch (BadJwtException failed) {
 			throw new InvalidBearerTokenException(failed.getMessage(), failed);
+		} catch (JwtException failed) {
+			throw new AuthenticationServiceException(failed.getMessage(), failed);
```

Comment:
I'm wondering if this is correct? When `AuthenticationServiceException` is thrown, it typically is caught by an `AbstractAuthenticationProcessingFilter`, which `BearerTokenAuthenticationFilter` is not.